Built-in v-model modifiers produce type errors with v3.1.2

### Environment

- Operating System: Linux
- Node Version:     v22.14.0
- Nuxt Version:     3.17.3
- CLI Version:      3.25.1
- Nitro Version:    2.11.12
- Package Manager:  npm@11.3.0
- Builder:          -
- User Config:      app, build, colorMode, compatibilityDate, debug, devtools, fonts, future, hooks, i18n, icon, imports, modules, nitro, routeRules, runtimeConfig, security, ssr, sourcemap, css, telemetry, vite
- Runtime Modules:  @nuxt/eslint@1.3.1, @pinia/nuxt@0.11.0, @vueuse/nuxt@13.2.0, @nuxtjs/i18n@9.5.4, nuxt-security@2.2.0, @nuxt/ui@3.1.2
- Build Modules:    -

### Is this bug related to Nuxt or Vue?

Nuxt

### Version

v3.1.2

### Reproduction

https://codesandbox.io/p/devbox/sharp-christian-kzdxt9?file=%2Fapp%2Fpages%2Findex.vue

### Description

Starting from v3.1.2, using the [built-in v-model-modifiers](https://vuejs.org/guide/essentials/forms.html#modifiers) produces type errors. They are still working tho, so its only a type issue.

Reproduction:
1. Open provided link
2. Run `pnpx nuxi typecheck`

### Additional context

_No response_

### Logs

```shell-script
➜  workspace git:(master) ✗ pnpx nuxi typecheck
Packages: +1
+
Progress: resolved 1, reused 1, downloaded 0, added 1, done
ℹ Nuxt Icon server bundle mode is set to local                                                                                                                                                        11:09:23 AM
✔ Nuxt Icon discovered local-installed 2 collections: lucide, simple-icons                                                                                                                            11:09:25 AM
ℹ Running with compatibility version 4                                                                                                                                                           nuxt 11:09:25 AM
app/pages/index.vue:8:22 - error TS2353: Object literal may only specify known properties, and 'modelModifiers' does not exist in type '{ readonly onBlur?: ((event: FocusEvent) => any) | undefined; readonly onChange?: ((event: Event) => any) | undefined; readonly "onUpdate:modelValue"?: ((payload: number) => any) | undefined; ... 24 more ...; modelValue?: number | undefined; } & VNodeProps & AllowedComponentProps & ComponentCustomProps'.

8       <UInput v-model.number="myNumber" type="number" />
                       ~~~~~~~

app/pages/index.vue:9:22 - error TS2353: Object literal may only specify known properties, and 'modelModifiers' does not exist in type '{ readonly onBlur?: ((event: FocusEvent) => any) | undefined; readonly onChange?: ((event: Event) => any) | undefined; readonly "onUpdate:modelValue"?: ((payload: string) => any) | undefined; ... 24 more ...; modelValue?: string | undefined; } & VNodeProps & AllowedComponentProps & ComponentCustomProps'.

9       <UInput v-model.lazy="myText" />
                       ~~~~~


Found 2 errors in the same file, starting at: app/pages/index.vue:8


 ERROR  Process exited with non-zero status (2)                                                                                                                                                        11:09:41 AM

    at R._waitForOutput (/root/.cache/pnpm/v3/tmp/dlx-4483/node_modules/.pnpm/nuxi@3.25.1/node_modules/nuxi/dist/chunks/main.mjs:508:13)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Object.run (/root/.cache/pnpm/v3/tmp/dlx-4483/node_modules/.pnpm/nuxi@3.25.1/node_modules/nuxi/dist/chunks/typecheck.mjs:91:9)
    at async runCommand$1 (/root/.cache/pnpm/v3/tmp/dlx-4483/node_modules/.pnpm/nuxi@3.25.1/node_modules/nuxi/dist/shared/nuxi.BlFGnQYG.mjs:1767:16)
    at async runCommand$1 (/root/.cache/pnpm/v3/tmp/dlx-4483/node_modules/.pnpm/nuxi@3.25.1/node_modules/nuxi/dist/shared/nuxi.BlFGnQYG.mjs:1758:11)
    at async runMain$1 (/root/.cache/pnpm/v3/tmp/dlx-4483/node_modules/.pnpm/nuxi@3.25.1/node_modules/nuxi/dist/shared/nuxi.BlFGnQYG.mjs:1896:7) 



 ERROR  Process exited with non-zero status (2)
```

--- a/src/runtime/components/Input.vue
+++ b/src/runtime/components/Input.vue
@@ -38,7 +38,6 @@ export interface InputProps extends UseComponentIconsProps {
   disabled?: boolean
   /** Highlight the ring color like a focus state. */
   highlight?: boolean
-  class?: any
   modelModifiers?: {
     string?: boolean
     number?: boolean
@@ -46,6 +45,7 @@ export interface InputProps extends UseComponentIconsProps {
     lazy?: boolean
     nullify?: boolean
   }
+  class?: any
   ui?: Input['slots']
 }
 

--- a/src/runtime/components/Input.vue
+++ b/src/runtime/components/Input.vue
@@ -43,6 +43,7 @@ export interface InputProps extends UseComponentIconsProps {
     string?: boolean
     number?: boolean
     trim?: boolean
+    lazy?: boolean
     nullify?: boolean
   }
   ui?: Input['slots']

--- a/src/runtime/components/Input.vue
+++ b/src/runtime/components/Input.vue
@@ -39,6 +39,12 @@ export interface InputProps extends UseComponentIconsProps {
   /** Highlight the ring color like a focus state. */
   highlight?: boolean
   class?: any
+  modelModifiers?: {
+    string?: boolean
+    number?: boolean
+    trim?: boolean
+    nullify?: boolean
+  }
   ui?: Input['slots']
 }
 
@@ -77,6 +83,7 @@ const props = withDefaults(defineProps<InputProps>(), {
 const emits = defineEmits<InputEmits<T>>()
 const slots = defineSlots<InputSlots>()
 
+// eslint-disable-next-line vue/no-dupe-keys
 const [modelValue, modelModifiers] = defineModel<T>()
 
 const appConfig = useAppConfig() as Input['AppConfig']

--- a/src/runtime/components/Textarea.vue
+++ b/src/runtime/components/Textarea.vue
@@ -35,7 +35,6 @@ export interface TextareaProps extends UseComponentIconsProps {
   autoresize?: boolean
   autoresizeDelay?: number
   disabled?: boolean
-  class?: any
   rows?: number
   maxrows?: number
   /** Highlight the ring color like a focus state. */
@@ -46,6 +45,7 @@ export interface TextareaProps extends UseComponentIconsProps {
     lazy?: boolean
     nullify?: boolean
   }
+  class?: any
   ui?: Textarea['slots']
 }
 

--- a/src/runtime/components/Textarea.vue
+++ b/src/runtime/components/Textarea.vue
@@ -43,6 +43,7 @@ export interface TextareaProps extends UseComponentIconsProps {
   modelModifiers?: {
     string?: boolean
     trim?: boolean
+    lazy?: boolean
     nullify?: boolean
   }
   ui?: Textarea['slots']

--- a/src/runtime/components/Textarea.vue
+++ b/src/runtime/components/Textarea.vue
@@ -40,6 +40,12 @@ export interface TextareaProps extends UseComponentIconsProps {
   maxrows?: number
   /** Highlight the ring color like a focus state. */
   highlight?: boolean
+  modelModifiers?: {
+    string?: boolean
+    number?: boolean
+    trim?: boolean
+    nullify?: boolean
+  }
   ui?: Textarea['slots']
 }
 
@@ -76,6 +82,7 @@ const props = withDefaults(defineProps<TextareaProps>(), {
 const emits = defineEmits<TextareaEmits<T>>()
 const slots = defineSlots<TextareaSlots>()
 
+// eslint-disable-next-line vue/no-dupe-keys
 const [modelValue, modelModifiers] = defineModel<T>()
 
 const appConfig = useAppConfig() as Textarea['AppConfig']

--- a/src/runtime/components/Textarea.vue
+++ b/src/runtime/components/Textarea.vue
@@ -42,7 +42,6 @@ export interface TextareaProps extends UseComponentIconsProps {
   highlight?: boolean
   modelModifiers?: {
     string?: boolean
-    number?: boolean
     trim?: boolean
     nullify?: boolean
   }

